packages/jest-cli/src/runJest.js

```diff
@@ -271,19 +272,8 @@ export default (async function runJest({
     collectHandles = collectNodeHandles();
   }
 
-  if (globalConfig.globalSetup) {
-    // $FlowFixMe
-    const globalSetup = require(globalConfig.globalSetup);
-    if (typeof globalSetup !== 'function') {
-      throw new TypeError(
-        `globalSetup file must export a function at ${
-          globalConfig.globalSetup
-        }`,
-      );
-    }
+  await runGlobalModules({allTests, globalConfig, moduleName: 'globalSetup'});
```

Comment:
I'm not sure about the "modules" terminology. How about using "hook"?
```js
await runGlobalHook({allTests, globalConfig, hook: 'globalSetup'});
```
Other than that looks good

Reply:
Thanks, hook sounds good, but it runs multiple `globalSetup`/`globalTeardown` functions, so maybe hooks ?
```js
await runGlobalHooks(...)
```

Reply:
Hm, `runGlobalHookForProjects`? Not sure if we need to get this explicit, but it communicates clearly what it does

Reply:
It also runs the `globalSetup`/`globalTeardown` for the main, project, so let's go with your suggestion 

```js
await runGlobalHook(...)
```